beps/0001-notifications-system/README.md

```diff
@@ -0,0 +1,255 @@
+  payload: JsonObject;
+}
+
+interface SignalService {
+  signal(signal: Signal): Promise<void>;
+}
+```
+
+### Frontend API
+
+The following frontend API is added as part of this proposal.
+
+#### `SignalApi`
+
+> TODO - we likely need a slightly different approach here, since APIs are lazy loaded.
+
+```ts
+interface SignalSubscriber {
+  unsubscribe(): void;
+}
+
+interface SignalApi {
+  subscribe(
+    channel: string,
+    listener: (payload: JsonObject) => void,
+  ): SignalSubscriber;
+}
+```
+
+> TODO: Add payload formats and API details
+
+<!--
+
+
+- A notification can be sent to any entity
+  - If the entity is an user, the notification is sent to that user
+  - If the entity is a group, the notification is sent to all users that are members of that group
+  - In other cases, the notification is sent to to the user or group in the `spec.owner` field
+- Notification frontend shows users their own notifications in its own page and the number of unread notifications
+  in the main menu item
+- Notifications can be set to `read` by the user and they are filtered out of the main view
+- Notifications can be saved for later use by the user
+- Each notification must have title, description and link
+- Each notification can also have icon or image as an URL
+- Notification frontend utilizes [Web Notification API](https://developer.mozilla.org/en-US/docs/Web/API/Notification)
+  to notify user for new notifications
+
+ -->
+
+## Release Plan
+
+The notification and signal plugins are released as two new plugins in the Backstage ecosystem. They will both start out in an experimental state.
+
+For the notification plugin to reach a stable release we much reach the following:
+
+- A stable notifications payload format.
+- A stable notifications receiver filter format.
+- The event broker must have at least one implementation that supports scaled deployments.
+
+For the signal plugin to reach a stable release we much reach the following:
+
+- A stable signal receiver filter format.
+- A stable signal channel API in the frontend.
+
+If any changes are required to the frontend framework to facilitate the implementation of notifications or signals, these will be released as experimental alpha features. They will stay in alpha until they are deemed stable enough, which must happen before a stable release of the notifications system.
+
+## Dependencies
+
+Since the signal plugin relies on the event broker for communication, it is a dependency for the notifications system as a whole. The event broker does not currently implement any transport for scaled deployments, which is a requirement for scaled deployments of the notification system.
+
+## Alternatives
+
+### Signal Plugin Separation
+
+One primary consideration for the notifications system was whether the notification plugin should set up its own frontend push connection rather than relying on the signal plugin. The signal plugin separation is done for three primary reasons:
```

Comment:
This is great stuff! I already have use cases for the signal plugin as it's own in my Q&A plugin